endpoint/estuary.go

```diff
@@ -0,0 +1,78 @@
+package endpoint
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/application-research/autoretrieve/filecoin"
+	"github.com/application-research/filclient"
+	"github.com/filecoin-project/go-address"
+	"github.com/ipfs/go-cid"
+)
+
+var (
+	ErrInvalidEndpointURL    = errors.New("invalid endpoint URL")
+	ErrEndpointRequestFailed = errors.New("endpoint request failed")
+	ErrEndpointBodyInvalid   = errors.New("endpoint body invalid")
+)
+
+type EstuaryEndpoint struct {
+	url string
+	fc  *filclient.FilClient
+}
+
+type retrievalCandidate struct {
+	Miner   address.Address
+	RootCid cid.Cid
+	DealID  uint
+}
+
+func NewEstuaryEndpoint(fc *filclient.FilClient, url string) *EstuaryEndpoint {
+	return &EstuaryEndpoint{
+		url: url,
+		fc:  fc,
+	}
+}
+
+func (ee *EstuaryEndpoint) FindCandidates(ctx context.Context, cid cid.Cid) ([]filecoin.RetrievalCandidate, error) {
```

Comment:
can you define the `Endpoint` interface that both  the estuary endpoint and the indexer endpoint conform to?